Evaluate-ModelarDB-Compression/main.py

```diff
@@ -0,0 +1,281 @@
+    real_column = real_column.to_numpy().astype(numpy.float32)
+    compressed_column = compressed_column.to_numpy()
+
+    # Initialize variables for computing metrics.
+    equal_values = 0
+    compared_values = 0
+
+    sum_difference = 0.0
+    sum_real_values = 0.0
+
+    max_actual_error = 0.0
+    max_actual_error_real_value = 0.0
+    max_actual_error_compressed_value = 0.0
+
+    ceiled_actual_errors = math.ceil(error_bound + 2) * [0]
+
+    # Compute metrics.
+    for real_value, compressed_value in zip(real_column, compressed_column):
+        if real_value == compressed_value:
+            equal_values += 1.0
+            difference = 0.0
+            actual_error = 0.0
+        else:
+            difference = real_value - compressed_value
+            actual_error = abs(difference / real_value)
+
+        compared_values += 1
+
+        sum_difference += difference
+        sum_real_values += real_value
+
+        if max_actual_error < actual_error:
+            max_actual_error = actual_error
+            max_actual_error_real_value = real_value
+            max_actual_error_compressed_value = compressed_value
+
+        try:
+            ceiled_actual_errors[math.ceil(actual_error)] += 1
+        except OverflowError:
+            print(
+                "ERROR: undefined error due to {} (real) and {} (compressed).".format(
+                    real_value, compressed_value
+                )
+            )
+            print()
+            return
+
+    # Compute and print the final result.
+    print("- Real Values: {}".format(len(real_column)))
+    print("- Compressed Values: {}".format(len(compressed_column)))
+    print("- Without Error: {}%".format(100 * (equal_values / compared_values)))
+    print("- Average Error: {}%".format(100 * abs(sum_difference / sum_real_values)))
+    print(
+        "- Maximum Error: {}% due to {} (real) and {} (compressed)".format(
+            max_actual_error,
+            max_actual_error_real_value,
+            max_actual_error_compressed_value,
+        )
+    )
+    print("- Error Ceil Histogram:", end="")
+    for ceiled_error_cound, count in enumerate(ceiled_actual_errors):
+        print(" {}% {} ".format(ceiled_error_cound, count), end="")
+    print("\n")
+
+
+def measure_data_folder_size_in_kib(data_folder):
+    du_output = subprocess.check_output(["du", "-k", "-d0", data_folder])
+    return int(du_output.split(b"\t")[0])
+
+
+def send_sigint_to_process(process):
+    process.send_signal(signal.SIGINT)
+
+    # Ensure process is fully shutdown.
+    while process.poll() is None:
+        time.sleep(10)
+
+    process.wait()
+
+    stderr = process.stderr.read()
+    if errors_occurred(stderr):
+        print_stream(stderr)
+        return True
+
+
+# Main Function.
+if __name__ == "__main__":
+    # Ensure the necessary arguments are provided.
+    if len(sys.argv) < 2:
+        print("usage: " + sys.argv[0] + "test_data.parquet error_bound*")
```

Comment:
Maybe use an [f-string](https://peps.python.org/pep-0498/) instead to make it a little more readable. This also applies to all other places in `main` where either `.format()` is used or `"string" + arg` is used.

Reply:
Fixed.